rust/protocol/src/protocol.rs

```diff
@@ -17,6 +20,44 @@ use uuid::Uuid;
 
 pub const CIPHERTEXT_MESSAGE_CURRENT_VERSION: u8 = 3;
 
+/// A [u8] describing the version of the message chain format to use when starting a chain.
+#[derive(Copy, Clone, Eq, PartialEq, Debug, num_enum::TryFromPrimitive, num_enum::IntoPrimitive)]
+#[repr(u8)]
+pub enum MessageVersion {
+    Version2 = 2,
+    /// **\[CURRENT\]**.
+    Version3 = CIPHERTEXT_MESSAGE_CURRENT_VERSION,
+}
```

Comment:
I think this would go the other way around: the constant should have enum type now. (Also, I'd just spell these as "V2" and "V3"; the enum name already has "Version" spelled out.)

Reply:
Renamed to `V2` and `V3`, and converted the constant to be enum type!